### PR TITLE
DEP Allow symfony 4

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripeltd/product-issues/issues/492

### CODE REVIEW

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
The [`Process`](https://github.com/silverstripe/silverstripe-fulltextsearch/blob/9de2764faaa63cb3acec9c8a5f249303cb756cec/src/Solr/Reindex/Handlers/SolrReindexImmediateHandler.php#L103) constructor now only accepts an array. Passing the command as a string will [throw a deprecation warning](https://github.com/symfony/process/blob/c2098705326addae6e6742151dfade47ac71da1b/Process.php#L146-L148).

**Comment by emteknetnz** [comment]
Updated it to pass an array

**symfony 3 array|string**
https://github.com/symfony/process/blob/3.4/Process.php#L135

**symfony 4 array**
https://github.com/symfony/process/blob/4.4/Process.php

**How symfony 3 handles arrays (implodes them)**
https://github.com/symfony/process/blob/3.4/Process.php#L285

File: src/Solr/Reindex/Handlers/SolrReindexImmediateHandler.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ protected function processGroup(
         $logger->info("Running '$cmd'");
 
         // Execute script via shell
-        $process = new Process($cmd);
+        $process = new Process([$cmd]);
```
Comment (maxime-rainville) [inline src/Solr/Reindex/Handlers/SolrReindexImmediateHandler.php:103]: The [Symphony doc](https://symfony.com/doc/current/components/process.html#usage) says to pass the parameters as array entries:
```php
$process = new Process(['ls', '-lsa']);
```

So it probably needs to be something like this:
```suggestion
        $process = new Process([
            $scriptTask, 
            "index={$indexClassEscaped}",
            "class={$class}",
            "group={$group}",
            "groups={$groups}",
            "variantstate={$statevar}",
            "verbose=1"
        ]);
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: src/Solr/Reindex/Handlers/SolrReindexImmediateHandler.php (hunk 1 above)
Comment (emteknetnz) [reply src/Solr/Reindex/Handlers/SolrReindexImmediateHandler.php]: Yes agree that would be better, have updated

**Review by maxime-rainville — APPROVED** [review]
Tested locally with both symfony 3 and 4. Works great.

**Comment by GuySartorelli** [comment]
@maxime-rainville Heya, just wondering if this got tested in dev mode, as well as test/live mode? I'm having an issue at the moment after an update that includes updating symfony from 3 to 4 - in dev mode the processes don't seem to be run, though replacing that with `shell_exec` does work.

**Pull request merged**